src/XMakeBuildEngine/Construction/ProjectMetadataElement.cs

```diff
@@ -131,17 +131,22 @@ internal void ChangeName(string newName)
 
         internal static void ValidateValidMetadataAsAttributeName(string name, string parentName, IElementLocation parentLocation)
         {
-            bool isKnownAttribute;
-            bool isValidMetadataNameInAttribute;
-
-            ProjectParser.CheckMetadataAsAttributeName(name, out isKnownAttribute, out isValidMetadataNameInAttribute);
-
-            if (isKnownAttribute || !isValidMetadataNameInAttribute)
+            if (!AttributeNameCanRepresentMetadata(name))
             {
                 ProjectErrorUtilities.ThrowInvalidProject(parentLocation, "InvalidMetadataAsAttribute", name, parentName);
             }
         }
 
+        internal static bool AttributeNameCanRepresentMetadata(string name)
```

Comment:
I find this name confusing, but I can't think of one that's unambiguously better. `AttributeIsMetadata`? `AttributeIsNotReserved`?

Fine to go with current if we can't think of something better.